src/form.ts

```diff
@@ -0,0 +1,173 @@
+const EXCLUDED_TAGS = /\b(?:file|submit|image|reset|button)\b/;
+
+type FormValue = { [ key: string ]: any };
+
+/**
+ * Fills in a DOM form using values from a JavaScript object. Note that fields not specified in the value object will be
+ * cleared.
+ *
+ * @param form the DOM form to set values in
+ * @param object an object containing values for each field in the form
+ *
+ * @example
+ * let formNode = document.getElementById('aForm');
+ * form.fromObject(formNode, { firstName: 'foo', lastName: 'bar' });
+ */
+export function fromObject(form: HTMLFormElement, object: FormValue): void {
+	const elements = form.elements;
+	for (let i = 0; i < elements.length; i++) {
+		const element = <HTMLInputElement> elements[i];
```

Comment:
TS won't compile that because `elements` isn't an array or a string. I could `Array.prototype.slice` it into an array, but that seems inefficient.